syntax = "proto3";
package ong.inventory;
import "google/protobuf/empty.proto";

enum Category {
  ROPA = 0;
  ALIMENTOS = 1;
  JUGUETES = 2;
  UTILES_ESCOLARES = 3;
}

message DonationItem {
  int32 id = 1;
  Category category = 2;
  string description = 3;
  int32 quantity = 4;
  bool deleted = 5;
  string created_at = 6;
  string created_by = 7;
  string updated_at = 8;
  string updated_by = 9;
}

message CreateDonationRequest {
  string actor_username = 1;
  Category category = 2;
  string description = 3;
  int32 quantity = 4;
}

message UpdateDonationRequest {
  string actor_username = 1;
  int32 id = 2;
  string description = 3;
  int32 quantity = 4;
}

message DonationIdRequest { string actor_username = 1; int32 id = 2; }
message DonationList { repeated DonationItem items = 1; }

service DonacionesService {
  rpc CreateDonation(CreateDonationRequest) returns (DonationItem);
  rpc UpdateDonation(UpdateDonationRequest) returns (DonationItem);
  rpc DeleteDonation(DonationIdRequest) returns (google.protobuf.Empty); // baja l√≥gica
  rpc ListDonations(google.protobuf.Empty) returns (DonationList);
}
